handlers/smshandler/adminapi/adminapi.go

```
@@ -0,0 +1,323 @@
+		return err
+	}
+	return ctx.Send([]byte(respOK), bearerstdapi.HTTPstatusCodeOK)
+}
+
+func addElection(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID, electionID types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if err := electionID.FromString(ctx.URLParam("electionid")); err != nil {
+		return err
+	}
+	user, err := storage.User(userID)
+	if err != nil {
+		return err
+	}
+	election := smshandler.UserElection{
+		ElectionID:        electionID,
+		RemainingAttempts: storage.MaxAttempts(),
+	}
+	user.Elections = append(user.Elections, election)
+	if err := storage.UpdateUser(user); err != nil {
+		return err
+	}
+	return ctx.Send([]byte(respOK), bearerstdapi.HTTPstatusCodeOK)
+}
+
+func addAttempt(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID, election types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if err := election.FromString(ctx.URLParam("election")); err != nil {
+		return err
+	}
+	if err := storage.IncreaseAttempt(userID, election); err != nil {
+		return err
+	}
+	return ctx.Send([]byte(respOK), bearerstdapi.HTTPstatusCodeOK)
+}
+
+func setConsumed(msg *bearerstdapi.BearerStandardAPIdata, ctx *httprouter.HTTPContext) error {
+	var userID, election types.HexBytes
+	if err := userID.FromString(ctx.URLParam("userid")); err != nil {
+		return err
+	}
+	if err := election.FromString(ctx.URLParam("election")); err != nil {
+		return err
+	}
+	consumed := ctx.URLParam("consumed") == "true" || ctx.URLParam("consumed") == "1"
+	user, err := storage.User(userID)
+	if err != nil {
+		return err
+	}
+	i := user.FindElection(election)
+	if i == -1 {
+		return fmt.Errorf("election not found for user %s", user.UserID)
+	}
+	user.Elections[i].Consumed = consumed
+	if err := storage.UpdateUser(user); err != nil {
```

If `consumed` is not changed the update is made anyways, better to return an error if trying to set the same value ?